Model/ResourceModel/ProductLabel.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Smile\ProductLabel\Model\ResourceModel;
+
+use Magento\Framework\DB\Select;
+use Magento\Framework\EntityManager\EntityManager;
+use Magento\Framework\EntityManager\MetadataPool;
+use Magento\Framework\Model\ResourceModel\Db\AbstractDb;
+use Magento\Framework\Model\ResourceModel\Db\Context;
+use Magento\Framework\Model\AbstractModel;
+use Smile\ProductLabel\Api\Data\ProductLabelInterface;
+
+/**
+ * Collection Resource Model Class: ProductLabel
+ *
+ * @category  Smile
+ * @package   Smile\ProductLabel
+ * @author    Houda EL RHOZLANE <hoelr@smile.fr>
+ * @copyright 2019 Smile
+ */
+class ProductLabel extends AbstractDb
+{
+
+    /**
+     * @var EntityManager
+     */
+    protected $entityManager;
+
+    /**
+     * @var MetadataPool
+     */
+    protected $metadataPool;
+
+    /**
+     * Resource initialization
+     *
+     * @param Context       $context
+     * @param EntityManager $entityManager
+     * @param MetadataPool  $metadataPool
+     * @param null          $connectionName
+     */
+    public function __construct(
+        Context $context,
+        EntityManager $entityManager,
+        MetadataPool $metadataPool,
+        $connectionName = null
+    )
+    {
+        parent::__construct($context, $connectionName);
+
+        $this->entityManager = $entityManager;
+        $this->metadataPool = $metadataPool;
+    }
+
+    protected function _construct()
+    {
+        $this->_init(
+            ProductLabelInterface::TABLE_NAME,
+            ProductLabelInterface::PRODUCTLABEL_ID);
+    }
+
+    public function getConnection()
+    {
+        $connectionName = $this->metadataPool->getMetadata(ProductLabelInterface::class)->getEntityConnectionName();
+
+        return $this->_resources->getConnectionByName($connectionName);
+    }
+
+    public function load(AbstractModel $object, $value, $field = null)
+    {
+        $objectId = $this->getObjectId($value, $field);
+
+        if ($objectId)
+            $this->entityManager->load($object, $objectId);
+
+        return $this;
+    }
+
+    public function save(AbstractModel $object)
+    {
+        $this->entityManager->save($object);
+
+        return $this;
+    }
+
+    public function delete(AbstractModel $object)
+    {
+        $this->entityManager->delete($object);
+
+        return $this;
+    }
+
+    protected function getObjectId($value, $field = null) {
```

Comment:
What is this method used for ? Probably not needed because we will not load product labels with another field than "id". It could be removed in my opinion, and the `load` method would be more easy .

Reply:
Done! You're right, i removed it